ibis/backends/base/sql/__init__.py

```diff
@@ -206,6 +204,13 @@ def to_pyarrow_batches(
 
         return pa.ipc.RecordBatchReader.from_batches(schema.to_pyarrow(), batches)
 
+    def _compile_udfs(self, expr: ir.Expr) -> Iterator[str]:
+        """Return an iterator of DDL strings, once for each UDFs contained within `expr`."""
+        if self.supports_python_udfs:
+            raise NotImplementedError(self.name)
+        return
+        yield
```

Comment:
This is expected to return a generator, and for backends that don't support UDFs it's a no-op. This appears to be one way to do it.

I suppose `return iter(())` could also work 🤷🏻

Reply:
I think we should make this method an abstract method on a mixin instead of these odd configuration settings and runtime checks.

I'll do that in a follow-up though.

Reply:
Is there a good reason to *require* this method return a generator (rather than leaving that backend implementation dependent)? If not, I'd change this to `Iterable[str]` and return [] by default. No change should be needed downstream consumers.

Reply:
Alternatively, do we expect the output of `list(_compile_udfs(...))` to be large enough that using a generator for memory reasons makes sense? Otherwise it may be nicer to always return a `list[str]`. This would remove the need for calling `consume` below.

Reply:
It seems wasteful to return a `list` when all I'm doing is iterating over it. The implementation is not meaningfully simpler with a list IMO except in the UDFs-not-supported case.

That said, definitely willing to change the types and simplify the udfs-not-supported case.

Reply:
Types fixed, udfs-not-supported case simplified.